src/request.ts

```diff
@@ -185,6 +185,11 @@ export class Request {
   public headers() {
     const headerAttr = this.methodDescriptor.headersAttr
     const headers = (this.requestParams[headerAttr] || {}) as Headers
```

Comment:
It's rather sad that the types here are lying, since we know this can be a function, as evidenced by the line after this.
The problem is that having functions for headers is a thing that's highly specific to mocks, and changing the type here will affect everything that calls `request.headers()` to have to treat possible function returns.

Reply:
IMHO this is one more case for not creating a Request between MockResource and MockRequest, but this afaict is a breaking change :T